Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/AsyncPDFRenderer.mm

```diff
@@ -133,15 +135,25 @@
         if (!protectedThis)
             return;
 
-        RefPtr plugin = protectedThis->m_plugin.get();
-        if (!plugin)
-            return;
-
-        protectedThis->m_pagePreviews.set(pagePreviewRequest.pageIndex, WTFMove(imageBuffer));
-        plugin->didGeneratePreviewForPage(pagePreviewRequest.pageIndex);
+        protectedThis->didCompletePagePreviewRender(WTFMove(imageBuffer), pagePreviewRequest);
     });
 }
 
+void AsyncPDFRenderer::didCompletePagePreviewRender(RefPtr<ImageBuffer>&& imageBuffer, const PagePreviewRequest& pagePreviewRequest)
```

Comment:
nit: Since we only need the `pageIndex`, maybe don't pass in all of `PagePreviewRequest`? Though I suppose that doesn't help with anything either way.

Reply:
I just prefer to pass along the same arguments we passed to the preview request.